coral-common/src/main/java/com/linkedin/coral/common/functions/FunctionReturnTypes.java

```diff
@@ -56,9 +57,18 @@ private FunctionReturnTypes() {
   public static final SqlReturnTypeInference ARRAY_OF_ARG0_TYPE =
       opBinding -> opBinding.getTypeFactory().createArrayType(opBinding.getOperandType(0), -1);
 
-  public static SqlReturnTypeInference arrayOfType(final SqlTypeName typeName) {
-    return opBinding -> opBinding.getTypeFactory().createArrayType(opBinding.getTypeFactory().createSqlType(typeName),
-        -1);
+  public static SqlReturnTypeInference arrayOfType(final SqlTypeName typeName, boolean elementsNullable) {
```

Comment:
Let us keep the old method with the old behavior for backward compatibility. We can have 2 signatures with one calling the other.

Reply:
Done. Added back the old API.